src/shell/boards/board-tab.tsx

```diff
@@ -50,51 +80,67 @@ export const AppBoardTab: FC<{ id: string; icon: string; title: string; fallback
 		(ev) => {
 			ev.stopPropagation();
 			closeBoard(id);
-			if (fallbackId) {
-				setCurrentBoard(fallbackId);
-			}
 		},
-		[fallbackId, id]
+		[id]
 	);
 
+	const textRef = useRef<HTMLDivElement>(null);
+	const [textOverflowing, setTextOverflowing] = useState(false);
+	const resizeObserverRef = useRef<ResizeObserver>();
+	useEffect(() => {
+		if (textRef.current && !resizeObserverRef.current) {
+			resizeObserverRef.current = new ResizeObserver((entries) => {
+				requestAnimationFrame(() => {
+					if (textRef.current) {
+						setTextOverflowing(textRef.current.offsetWidth < textRef.current.scrollWidth);
+					}
+				});
+			});
+			resizeObserverRef.current.observe(textRef.current);
+		}
+
+		return (): void => {
+			resizeObserverRef.current?.disconnect();
+		};
+	}, [textRef]);
+
 	return (
-		<Container orientation="row" width="fit" maxWidth="100%">
-			{current !== id && <VerticalDivider />}
+		<>
+			{!firstTab && <VerticalDivider />}
 			<TabContainer
+				wrap={'nowrap'}
+				orientation="row"
+				mainAlignment={'flex-start'}
+				onClick={onClick}
 				data-testid={`board-tab-${id}`}
+				padding={{ horizontal: 'extrasmall' }}
 				active={current === id}
-				padding={{ all: 'extrasmall' }}
 			>
-				<Row
-					height="100%"
-					onClick={onClick}
-					takeAvailableSpace
-					mainAlignment="flex-start"
-					wrap="nowrap"
+				<TabIcon icon={icon} size="large" />
+				{/* <Tooltip label={title} placement="top" maxWidth="700px"> */}
+				<CustomText
+					ref={textRef}
+					size="medium"
+					weight="regular"
+					color={current === id ? 'text' : 'secondary'}
+					overflowing={textOverflowing}
+					className="TabText"
```

Comment:
css convention would want the kebab-case for class names. I think it could be also usefule to distinguish it from the classes generated and set by styled-components